src/library/PiaFormat.ts

```diff
@@ -0,0 +1,94 @@
+import dayjs from "dayjs";
+
+enum PIATypesEnum {
+  piaString = "string",
+  piaDate = "date",
+  piaMonthYear = "monthyear",
+  piaInt = "int",
+}
+
+const piaLineParsers = {
+  "01": (contents: string) => {
+    const endPositions = {
```

Comment:
This approach is super clever, but in our case we don't want to use a general parsing approach because we know exactly what types we want back. It would be better to write a separate parsing function for each type, and then use them in the line parsers.

For example:
```
parsePiaDate(piaValue: string): Date {
  if (value.length !== 8) {
        console.warn("piaFormat date is invalid length");
  }
  /* dayjs will use local timezone */
  /* starts in mmddyyyy, dayjs takes yyyy-mm-dd */
  return dayjs(
    [value.slice(4, 8), value.slice(0, 2), value.slice(2, 4)].join("-")
  ).toDate();
}

function parsePiaNumber(piaValue: string): number {...}
// etc...
```

Then here in the parsers, we do this:
```
"01": (contents: string): Partial<PiaData> => {
  return {
    ssn: parsePiaString(contents.slice(3, 11)),
    birthDate: parsePiaData(contents.slice(13, 20)),
    sex: parsePiaSex(contents.slice(12, 13)),
  }
}
```